psutil/_psutil_linux.c

```diff
@@ -541,14 +541,15 @@ psutil_net_if_stats(PyObject* self, PyObject* args) {
     close(sock);
     py_retlist = Py_BuildValue("[Oiii]", py_is_up, duplex, speed, mtu);
     if (!py_retlist)
-        goto error;
+        goto error_after_close;
     Py_DECREF(py_is_up);
     return py_retlist;
 
 error:
-    Py_XDECREF(py_is_up);
-    if (sock != 0)
+    if (sock != -1)
         close(sock);
+error_after_close:
+    Py_XDECREF(py_is_up);
```

Comment:
Well, once a socket is closed - the file descriptor is available for use elsewhere. And thus a double close could end up closing that - which may cause a problem elsewhere. If you don't like multiple goto handlers, you might prefer moving the close(sock) on line 541 down to 545 before the return. That way there is a close in every path in the function - and no double close.